script.user.js

```diff
@@ -182,6 +182,48 @@ function loadEmoteSetEditor() {
 }
 
 
+// Tab completion
+function applyComplete() {
+    const textInput = document.getElementById("gcInput");
+
+    function getEmoteCompletionList(partialName) {
+        const emoteNames = Object.keys(emotes);
+        const lowerPartialName = partialName.toLowerCase();
+        return emoteNames.filter((name) =>
+                                 name.toLowerCase().startsWith(lowerPartialName)
+                                );
+    }
+
+    function onTabPressed(event) {
```

Comment:
This also doesn't need to be nested. Just use `const textInput = event.currentTarget`.